ibis/expr/operations/relations.py

```diff
@@ -84,6 +85,57 @@ def change_name(self, new_name):
         return type(self)(new_name, self.args[1], self.source)
 
 
+@rlz.validator
+def uri(arg, **kwargs):
+    arg = rlz.instance_of(str, arg)
+    uri = urllib.parse.urlparse(arg)
+    if not uri.scheme:
+        raise ValueError(f"Missing URI scheme in '{arg}'")
+    return arg
+
+
+@public
+class FileFormat(Node):
+    """
+    Base class representing the format of files in a FileTable.
+    """
+
+
+@public
+class ArrowFileFormat(FileFormat):
+    """
+    Files of Arrow IPC format.
+    """
+
+
+@public
+class OrcFileFormat(FileFormat):
+    """
+    Files of ORC format.
+    """
+
+
+@public
+class ParquetFileFormat(FileFormat):
+    """
+    Files of Parquet format.
+    """
+
+
+@public
+class FileTable(TableNode, sch.HasSchema):
+    """
+    A table sourced from a files on a local/remote filesystem.
+
+    More common in systems shaped like 'query engines' (e.g. Pandas,
+    Acero, DuckDB).
+    """
+
+    items = rlz.tuple_of(uri)
+    schema = rlz.instance_of(sch.Schema)
+    file_format = rlz.instance_of(FileFormat)
```

Comment:
Do we plan to extend the formats with additional descriptors? Otherwise shall we use an enum instead?

Reply:
Hmm, Parquet might want to be able to support some options (dictionary-encoding?). If/when we add CSV (maybe I should add that now?), we'd also want options for things like delimiters and headers, at the very least.